Library/ViewModels/LandingPageViewModelTests.swift

```diff
@@ -6,17 +6,41 @@ import UIKit
 import XCTest
 
 internal final class LandingPageViewModelTests: TestCase {
-  fileprivate let landingPageCards = TestObserver<[LandingPageCardType], Never>()
-  fileprivate let numberOfPages = TestObserver<Int, Never>()
-  fileprivate let viewModel: LandingPageViewModelType = LandingPageViewModel()
+  private let dismissViewController = TestObserver<(), Never>()
+  private let landingPageCards = TestObserver<[LandingPageCardType], Never>()
+  private let numberOfPages = TestObserver<Int, Never>()
+  private let viewModel: LandingPageViewModelType = LandingPageViewModel()
 
   override func setUp() {
     super.setUp()
 
+    self.viewModel.outputs.dismissViewController.observe(self.dismissViewController.observer)
     self.viewModel.outputs.landingPageCards.observe(self.landingPageCards.observer)
     self.viewModel.outputs.numberOfPages.observe(self.numberOfPages.observer)
   }
 
+  func testDismissViewController_OnButtonTap() {
+    self.dismissViewController.assertDidNotEmitValue()
+
+    self.viewModel.inputs.ctaButtonTapped()
+
+    self.dismissViewController.assertValueCount(1)
+  }
+
+  func testUserDefaultsUpdates_OnButtonTap() {
+    let userDefaults = MockKeyValueStore()
+
+    withEnvironment(currentUser: nil, userDefaults: userDefaults) {
```

Comment:
Should we also test that the ubiquitous store value has been set?

Reply:
Actually, come to think of it, should we even be setting the ubiquitous store value? From my understanding, the ubiquitous store will persist across app installs, which means if the user sees the landing page, then deletes the app, they will never see the landing page again even when they re-install the app. Is this desired behavior?

Reply:
The experiment briefing says that the Landing Page should be only presented on the first session of the user. I guess this is another decision for product. Will keep you updated :)

Reply:
I removed the piece of code that persists the value on the ubiquitous store. Now we are considering UserDefaults.